pkg/auth/requests/service_account.go

```diff
@@ -0,0 +1,169 @@
+	mgmtv3 "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/types/config"
+	"github.com/rancher/steve/pkg/auth"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apiserver/pkg/authentication/serviceaccount"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type (
+	restConfigGetter  func(cluster *mgmtv3.Cluster, context *config.ScaledContext, secretLister corev1.SecretLister) (*rest.Config, error)
+	authClientCreator func(clusterID string) (kubernetes.Interface, error)
+)
+
+// ServiceAccountAuth is an authenticator that authenticates requests using the downstream service account's JWT.
+type ServiceAccountAuth struct {
+	scaledContext             *config.ScaledContext
+	clusterLister             mgmtv3.ClusterLister
+	secretLister              corev1.SecretLister
+	restConfigGetter          restConfigGetter
+	authClientCreator         authClientCreator
+	clusterProxyConfigsGetter controllers.ClusterProxyConfigCache
+}
+
+// NewServiceAccountAuth creates a new instance of ServiceAccountAuth.
+func NewServiceAccountAuth(
+	scaledContext *config.ScaledContext,
+	restConfigGetter restConfigGetter,
+) auth.Authenticator {
+	return &ServiceAccountAuth{
+		scaledContext:    scaledContext,
+		clusterLister:    scaledContext.Management.Clusters("").Controller().Lister(),
+		secretLister:     scaledContext.Core.Secrets("").Controller().Lister(),
+		restConfigGetter: restConfigGetter,
+		authClientCreator: func(clusterID string) (kubernetes.Interface, error) {
+			return scaledContext.Wrangler.MultiClusterManager.K8sClient(clusterID)
+		},
+		clusterProxyConfigsGetter: scaledContext.Wrangler.Mgmt.ClusterProxyConfig().Cache(),
+	}
+}
+
+// Authenticate the request using the downstream service account's JWT.
+func (t *ServiceAccountAuth) Authenticate(req *http.Request) (user.Info, bool, error) {
+	// Basic checks to see if we even need to proceed
+	info, hasAuth := request.UserFrom(req.Context())
+	if info.GetName() != "system:cattle:error" {
+		return info, hasAuth, nil
+	}
+	clusterID := mux.Vars(req)["clusterID"]
+	if clusterID == "" {
+		return info, hasAuth, fmt.Errorf("no clusterID found in request")
+	}
+	// Check the cluster setting value to determine whether we will continue the auth process
+	settings, err := t.clusterProxyConfigsGetter.List(clusterID, labels.NewSelector())
+	if err != nil {
+		logrus.Debugf("rejecting downstream proxy request for %s, unable to fetch ClusterProxySettings object for cluster", req.URL.Path)
+		return &user.DefaultInfo{}, false, nil
+	}
+	if settings == nil {
+		logrus.Debugf("rejecting downstream proxy request for %s, no ClusterProxySettings object exists for cluster", req.URL.Path)
+		return &user.DefaultInfo{}, false, nil
+	}
+	if len(settings) > 1 {
```

Comment:
I don't see any tests for this?

Reply:
There's a companion PR, for webhook, that enforces the "only 1 settings object" rule.  We could likely remove the check here now that the webhook is in place.